File: src/africa/semicolon/MyersBriggs.java
package africa.semicolon;

import java.util.Arrays;
import java.util.Scanner;

public class MyersBriggs {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[][] extroversionQuestions = new String[5][2];
        String[][] sensingQuestions = new String[5][2];
        String[][] thinkingQuestions = new String[5][2];
        String[][] judgingQuestions = new String[5][2];

        extroversionQuestions[0][0] = "expend energy, enjoy groups";
        extroversionQuestions[0][1] = "conserve energy, enjoy one-on-one";
        extroversionQuestions[1][0] = "more outgoing, think out loud";
        extroversionQuestions[1][1] = "more reserved, think to yourself";
        extroversionQuestions[2][0] = "seek many tasks, public activities, interaction with others";
        extroversionQuestions[2][1] = "seek private, solitary activities with quiet to concentrate";
        extroversionQuestions[3][0] = "external, communicative, express yourself";
        extroversionQuestions[3][1] = "internal, reticent, keep to yourself";
        extroversionQuestions[4][0] = "active, initiate";
        extroversionQuestions[4][1] = "reflective, deliberate";

        sensingQuestions[0][0] = "interpret literally";
        sensingQuestions[0][1] = "look for meaning and possibilities";
        sensingQuestions[1][0] = "practical, realistic, experiential";
        sensingQuestions[1][1] = "imaginative, innovative, theoretical";
        sensingQuestions[2][0] = "standard, usual, conventional";
        sensingQuestions[2][1] = "different, novel, unique";
        sensingQuestions[3][0] = "focus on here-and-now";
        sensingQuestions[3][1] = "look to the future, global perspective, \"big picture\"";
        sensingQuestions[4][0] = "facts, things, \"what is\"";
        sensingQuestions[4][1] = "ideas, dreams, \"what could be,\" philosophical";

        thinkingQuestions[0][0] = "logical, thinking, questioning";
        thinkingQuestions[0][1] = "empathetic, feeling, accommodating";
        thinkingQuestions[1][0] = "candid, straight forward, frank";
        thinkingQuestions[1][1] = "tactful, kind, encouraging";
        thinkingQuestions[2][0] = "firm, tend to criticize, hold the line";
        thinkingQuestions[2][1] = "gentle, tend to appreciate, conciliate";
        thinkingQuestions[3][0] = "tough-minded, just";
        thinkingQuestions[3][1] = "tender-hearted, merciful";
        thinkingQuestions[4][0] = "matter of fact, issue-oriented";
        thinkingQuestions[4][1] = "sensitive, people-oriented, compassionate";

        judgingQuestions[0][0] = "organized, orderly";
        judgingQuestions[0][1] = "flexible, adaptable";
        judgingQuestions[1][0] = "plan, schedule";
        judgingQuestions[1][1] = "unplanned, spontaneous";
        judgingQuestions[2][0] = "regulated, structured”";
        judgingQuestions[2][1] = "easygoing, “live” and “let live”";
        judgingQuestions[3][0] = "preparation, plan ahead";
        judgingQuestions[3][1] = "go with the flow, adapt as you go";
        judgingQuestions[4][0] = "control, govern";
        judgingQuestions[4][1] = "latitude, freedom";

        int[] extroversion = new int[5];
        int[] sensing = new int[5];
        int[] thinking = new int[5];
        int[] judging = new int[5];

        StringBuilder result = new StringBuilder();
        int questionNumber = 1;
        for (int i = 0; i < extroversionQuestions.length; i++) {
            System.out.printf("Question %d%n", questionNumber++);
            System.out.println("A. " + extroversionQuestions[i][0]);
            System.out.println("B. " + extroversionQuestions[i][1]);
            System.out.println("Pick an option: A or B");
            String option = scanner.nextLine();
            if(option.equalsIgnoreCase("A")) extroversion[i] = 1;

            System.out.printf("Question %d%n", questionNumber++);
            System.out.println("A. " + sensingQuestions[i][0]);
            System.out.println("B. " + sensingQuestions[i][1]);
            System.out.println("Pick an option: A or B");
            option = scanner.nextLine();
            if(option.equalsIgnoreCase("A")) sensing[i] = 1;

            System.out.printf("Question %d%n", questionNumber++);
            System.out.println("A. " + thinkingQuestions[i][0]);
            System.out.println("B. " + thinkingQuestions[i][1]);
            System.out.println("Pick an option: A or B");
            option = scanner.nextLine();
            System.out.printf("Question %d%n", questionNumber++);
            if(option.equalsIgnoreCase("A")) thinking[i] = 1;

            System.out.println("A. " + judgingQuestions[i][0]);
            System.out.println("B. " + judgingQuestions[i][1]);
            System.out.println("Pick an option: A or B");
            option = scanner.nextLine();
            if(option.equalsIgnoreCase("A")) judging[i] = 1;

        }

//        System.out.printf("|%5s | %3s | %3s | %3s | %3s | %3s | %3s | %3s | %3s | %3s | %3s | %3s |", " ", "A", "B",
//                " ", "A", "B", " ", "A", "B", " ", "A", "B");
//        questionNumber = 1;
//        for (int i = 0; i < extroversion.length; i++) {
//
//        }

        if(sum(extroversion) < 3){
            result.append("I");
        }else {
            result.append("E");
        }

        if(sum(sensing) < 3){
            result.append("N");
        }else {
            result.append("S");
        }

        if(sum(thinking) < 3){
            result.append("F");
        }else {
            result.append("T");
        }

        if(sum(judging) < 3){
            result.append("P");
        }else {
            result.append("J");
        }


        System.out.println(Arrays.toString(extroversion));
        System.out.println(Arrays.toString(sensing));
        System.out.println(Arrays.toString(thinking));
        System.out.println(Arrays.toString(judging));

        System.out.println(result);
    }

    public static int sum(int[] intArrays){
        int sum = 0;
        for(int number : intArrays) sum += number;
        return sum;
    }
}

File: src/africa/semicolon/python/myers_briggs.py
questionnaire = [
    ["expend energy, enjoy groups", "conserve energy, enjoy one-on-one"],
    ["interpret literally", "look for meaning and possibilities"],
    ["logical, thinking, questioning", "empathetic, feeling, accommodating"],
    ["organized, orderly", "flexible, adaptable"],
    ["more outgoing, think out loud", "more reserved, think to yourself"],
    ["practical, realistic, experiential", "imaginative, innovative, theoretical"],
    ["candid, straight forward, frank", "tactful, kind, encouraging"],
    ["plan, schedule", "unplanned, spontaneous"],
    ["seek many tasks, public activities, interaction with others", "seek private, solitary activities "
                                                                    "with quiet to concentrate"],
    ["standard, usual, conventional", "different, novel, unique"],
    ["firm, tend to criticize, hold the line", "gentle, tend to appreciate, conciliate"],
    ["regulated, structured”", "easygoing, “live” and “let live”"],
    ["external, communicative, express yourself", "internal, reticent, keep to yourself"],
    ["focus on here-and-now", "look to the future, global perspective, \"big picture\""],
    ["tough-minded, just", "tender-hearted, merciful"],
    ["active, initiate", "reflective, deliberate"],
    ["preparation, plan ahead", "go with the flow, adapt as you go"],
    ["facts, things, \"what is\"", "ideas, dreams, \"what could be,\" philosophical"],
    ["matter of fact, issue-oriented", "sensitive, people-oriented, compassionate"],
    ["control, govern", "latitude, freedom"]
]


def get_option_a_or_b(response_option):
    if response_option.upper() == "A" or response_option.upper() == "B":
        return response_option
    else:
        raise ValueError("Wrong choice; choose A or B")


def get_option():
    while True:
        try:
            response_option = get_option_a_or_b(input())
            return response_option
        except ValueError as err:
            print("\033[91m" + str(err))


def place_checkmark(num, position):
    return chr(10003) if (num == 1 and position == 1) or (num == 0 and position == 2) else ""


responses = [0] * 20
option = ""
i = 0
for question in questionnaire:

    print(f"Question {i + 1}")
    print(f"A. {question[0]}")
    print(f"B. {question[1]}")
    print("Pick an option: A or B")
    option = get_option()
    if option.upper() == "A":
        responses[i] = 1
    i += 1

personality = [["E", "I"], ["S", "N"], ["T", "F"], ["J", "P"]]
options_count = [[0] * 2 for _ in range(4)]
personality_type = ""
for i in range(len(responses) // 5):
    total = 0
    for j in range(i, len(responses), 4):
        total += responses[j]

    options_count[i][0] = total
    options_count[i][1] = 5 - total
    if total >= 3:
        personality_type += personality[i][0]
    else:
        personality_type += personality[i][1]

print(responses)
print("\nYour choice at a glance\n")
print("|%5s | %3s | %3s | %3s | %3s | %3s | %3s | %3s | %3s | %3s | %3s | %3s |" % (" ", "A", "B",
                                                                                    " ", "A", "B", " ", "A", "B", " ",
                                                                                    "A", "B"))
print("%s" % ("-" * 74))

for i in range(0, len(responses), 4):
    print("|%5d | %3s | %3s | %3d | %3s | %3s | %3d | %3s | %3s | %3d | %3s | %3s |" % (
        i + 1, place_checkmark(responses[i], 1), place_checkmark(responses[i], 2),
        i + 2, place_checkmark(responses[i + 1], 1), place_checkmark(responses[i + 1], 2), i + 3,
        place_checkmark(responses[i + 2], 1), place_checkmark(responses[i + 2], 2), i + 4,
        place_checkmark(responses[i + 3], 1), place_checkmark(responses[i + 3], 2)))

print("%s" % ("-" * 74))
print("|%5s | %3d | %3d | %3s | %3d | %3d | %3s | %3d | %3d | %3s | %3d | %3d |" % (
    "TOTAL", options_count[0][0], options_count[0][1],
    " ", options_count[1][0], options_count[1][1], " ", options_count[2][0], options_count[2][1], " ",
    options_count[3][0],
    options_count[3][1]))
print("%s" % ("-" * 74))
print("|%5s | %3s | %3s | %3s | %3s | %3s | %3s | %3s | %3s | %3s | %3s | %3s |" % (" ", "E", "I",
                                                                                    " ", "S", "N", " ", "T", "F", " ",
                                                                                    "J", "P"))

print("\nYour personality type is " + personality_type)
print("For your personality interpretation, visit : ")
print("https://www.truity.com/personality-type/" + personality_type)
print("For all personality types visit: ")
print("https://www.truity.com/page/16-personality-types-myers-briggs")
